addons/account_edi_facturx/models/account_edi_format.py

```diff
@@ -192,17 +192,19 @@ def _import_facturx(self, tree, invoice):
                                        account_predictive_bills_disable_prediction=True)) as invoice_form:
             # Partner (first step to avoid warning 'Warning! You must first select a partner.').
             partner_type = invoice_form.journal_id.type == 'purchase' and 'SellerTradeParty' or 'BuyerTradeParty'
-            elements = tree.xpath('//ram:' + partner_type + '/ram:SpecifiedTaxRegistration/ram:ID', namespaces=tree.nsmap)
-            partner = elements and self.env['res.partner'].search([('vat', '=', elements[0].text)], limit=1)
-            if not partner:
-                elements = tree.xpath('//ram:' + partner_type + '/ram:Name', namespaces=tree.nsmap)
-                partner_name = elements and elements[0].text
-                partner = elements and self.env['res.partner'].search([('name', 'ilike', partner_name)], limit=1)
-            if not partner:
-                elements = tree.xpath('//ram:' + partner_type + '//ram:URIID[@schemeID=\'SMTP\']', namespaces=tree.nsmap)
-                partner = elements and self.env['res.partner'].search([('email', '=', elements[0].text)], limit=1)
-            if partner:
-                invoice_form.partner_id = partner
+            invoice_form.partner_id = self._retrieve_partner(
+                name=self._find_value('//ram:' + partner_type + '/ram:Name', tree, namespaces=tree.nsmap),
+                mail=self._find_value('//ram:' + partner_type + '//ram:URIID[@schemeID=\'SMTP\']', tree, namespaces=tree.nsmap),
```

Comment:
```suggestion
                mail=self._find_value(f"//ram:{partner_type}//ram:URIID[@schemeID='SMTP']", tree, namespaces=tree.nsmap),
```

Didn't review functionally as @smetl is more aware than me, but just this suggestion :)

Reply:
Arf too late... I like this suggestion... :/

Reply:
Done in the fw-port since there was a conflict anyway.